mirage/awa_mirage.ml

```diff
@@ -107,32 +135,56 @@ module Make (F : Mirage_flow.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK) =
       end
     | Error () -> match t.state with
       | `Error e -> Lwt.return (Error e)
-      | `Eof -> Lwt.return (Ok `Eof)
-      | `Active _ -> assert false
+      | `Closed | `Read_closed _ | `Active _ | `Write_closed _ -> Lwt.return (Ok `Eof)
 
   let close t =
-    (* TODO ssh session teardown (send some protocol messages) *)
-    FLOW.close t.flow >|= fun () ->
-    t.state <- `Eof
+    (match t.state with
+     | `Active ssh | `Read_closed ssh | `Write_closed ssh ->
+       let ssh, msg = Awa.Client.close ssh in
+       t.state <- inject_state ssh t.state;
+       writev_flow t (Option.to_list msg) >|= fun _ ->
+       t.state <- `Closed
```

Comment:
Here I wonder if we should set the state to `` `Closed`` before sending a close message?? So we don't send multiple `Channel_close` messages.

Reply:
Where's the "multiple close messages" being sent?

Reply:
It would happen if `close` is called twice and while the first `writev_flow` is running the other `close` call is scheduled and does another `writev_flow`

Reply:
I see what you mean. done in 12040f4